src/library_webgl.js

```diff
@@ -739,6 +739,58 @@ for (/**@suppress{duplicate}*/var i = 0; i < {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
       disableHalfFloatExtensionIfBroken(ctx);
 #endif
 
+      // If end user enables *glGetProcAddress() functionality, then we must filter out
+      // all future WebGL extensions from being passed to the user, and only restrict to advertising
+      // extensions that the *glGetProcAddress() function knows to handle.
+#if GL_ENABLE_GET_PROC_ADDRESS
```

Comment:
What about if the user calls `glGetString(GL_EXTENSIONS)` but they don't set `GL_ENABLE_GET_PROC_ADDRESS`?   Is it OK that they see the full unfiltered list in this case?

Similarly `emscripten_webgl_get_supported_extensions`.. if `GL_ENABLE_GET_PROC_ADDRESS` is not set what do we expect it to return?

How about GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS which is enabled by default.. do we want to see the full unfiltered list by default?

How about the existing wrapping of getSupportedExtensions by GL_DISABLE_HALF_FLOAT_EXTENSION_IF_BROKEN?  Does that wrapper simply layer in top of this one?

Reply:
I think it should be OK to expose the unfiltered list in all cases when GetProcAddress is not enabled, because any code that uses an entry point Emscripten doesn't implement will cause a linker error. (And any extension that doesn't have entry points is OK.) It is a little unfortunate that they will have to `#ifdef __EMSCRIPTEN__` though instead of `#ifdef GL_EXT_clip_control` which many codebases will already have.

Come to think of it, it must be spec-valid in OpenGL for an implementation to list extensions that aren't available in the header - because you can use an older header with a newer driver. It's less guaranteed that applications would be robust to this though.

> GL_DISABLE_HALF_FLOAT_EXTENSION_IF_BROKEN

That wrapper shadows `WebGLRenderingContext.prototype.getSupportedExtensions` with `ctx.getSupportedExtensions` so I think it will apply first (at the JS level) and then this patch will apply on top.

Reply:
> It's less guaranteed that applications would be robust to this though.

Er, my point is, Emscripten could remove them from the header instead of filtering them out, and that would be a less robust but technically valid fix (that would work for Filament at least in this case).

Actually it would be more robust than I thought - if we did this, we'd be disabling not just the static entrypoint `glClipControlEXT`, but also `PFNGLCLIPCONTROLEXTPROC` which should be used by any program that's using GetProcAddress on that function. Example: https://github.com/google/filament/blob/b007e9137eb029e256675eceb5fd52e6b873cecd/filament/backend/src/opengl/gl_headers.h#L132